packages/transaction-controller/src/TransactionController.ts

```diff
@@ -767,17 +767,23 @@ export class TransactionController extends BaseControllerV1<
       validateGasValues(gasValues);
     }
 
-    log('Creating cancel transaction', transactionId, gasValues);
-
+    await this.pendingTransactionTracker.checkTransactions();
```

Comment:
The other downside of this is we're checking all pending transactions, so in the unlikely event you have 10 pending transactions for example, are we meaningfully delaying the submit and making this scenario more likely?